### PR TITLE
 Codebase Cleanup and Enhancement

### PR DESCRIPTION
**Pull request by xiaguan**
This pull request aims to improve the cleanliness and overall quality of our codebase. The changes made can be categorized into three main areas:

Code Quality Improvement: I have configured rustfmt to ensure better code quality. This will help us maintain a consistent coding style across the project. Additionally, I have incorporated a spell-check tool to catch and correct any spelling errors in our code and comments. This will enhance readability and reduce potential confusion.

Logging Enhancement: I have replaced `env_logger` with `tracing`. This change will provide us with a more robust and convenient logging experience. `tracing` offers structured, context-aware, and asynchronous-friendly logging, which will be beneficial for our project.

S3Metadata Log Cleanup: I have tidied up the logs printed by S3Metadata. This cleanup will make the logs more readable and easier to understand, which will be helpful for debugging and tracking.

### CODE REVIEW

File: src/async_fuse/fuse/abi_marker.rs [hunk 1]
```diff
@@ -8,8 +8,8 @@ use std::{mem, slice};
 ///
 /// [`FuseAbiData`] can not be implemented for ZSTs.
 ///
-/// [`FuseAbiData`] can be implemented for DSTs but there is no way to construct a custom DST reference.
-///
+/// [`FuseAbiData`] can be implemented for DSTs but there is no way to construct
+/// a custom DST reference.
```
[the remark this answers is not in the document]
Comment (xiaguan) [reply src/async_fuse/fuse/abi_marker.rs:12]: I don't think it's an error. We need to limit the length of our comments to a single line.

[the remark this answers is not in the document]
Comment (rogercloud) [reply src/async_fuse/fuse/abi_marker.rs:12]: Is it formatted by the tool?

[the remark this answers is not in the document]
Comment (xiaguan) [reply src/async_fuse/fuse/abi_marker.rs:12]: > Is it formatted by the tool?

Yes

File: Cargo.toml [hunk 2]
```diff
@@ -65,6 +65,9 @@ tokio = { version = "1.19.2", features = ["full"] }
 # The version of smol should be same as the etcd-client used
 smol = "1.2.4"
 etcd-client = "0.11.1"
+tracing = "0.1"
```
Comment (rogercloud) [inline Cargo.toml:68]: Shall we clean other log libs?

File: rustfmt.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+format_code_in_doc_comments = true
+format_macro_bodies = true
+format_macro_matchers = true
+normalize_comments = true
+normalize_doc_attributes = true
+imports_granularity = "Module"
```
Comment (rogercloud) [inline rustfmt.toml:6]: I prefer `crate`, it's less verbose.

File: src/async_fuse/memfs/cache/mod.rs [hunk 4]
```diff
@@ -325,27 +331,25 @@ impl GlobalCache {
...
-                {
-                    Some(global_index)
-                } else {
-                    None
-                }
-            } else {
-                None
-            }
-            */
+            // if let Some(bucket) = bucket_opt {
```
Comment (rogercloud) [inline src/async_fuse/memfs/cache/mod.rs:334]: Just remove it.

File: src/async_fuse/memfs/cache/mod.rs [hunk 5]
```diff
@@ -407,26 +411,24 @@ impl GlobalCache {
...
-                {
-                    IoMemBlock::new(Some(block.clone()), 0, block_size)
-                } else {
-                    IoMemBlock::new(None, 0, 0)
-                }
-            } else {
-                IoMemBlock::new(None, 0, 0)
-            }
-            */
+            // if let Some(bucket) = bucket_opt {
```
Comment (rogercloud) [inline src/async_fuse/memfs/cache/mod.rs:414]: Same as above. And if we really want to comment some code and keep the indent, is there any way to work around.

File: src/async_fuse/memfs/dist/etcd.rs [hunk 6]
```diff
@@ -79,26 +81,24 @@ pub async fn register_volume(
...
-        }
-
-        node_set
-    } else {
-        let mut hash = HashSet::new();
-        hash.insert(node_id.to_owned());
-        hash
-    };
-    */
+    // let new_volume_node_list = if let Some(node_list) = volume_node_list {
```
Comment (rogercloud) [inline src/async_fuse/memfs/dist/etcd.rs:84]: Same as above.

**Review by rogercloud — COMMENTED** [review]
For the code commented, if we don't need it anymore, just remove it. If we need to keep it as an example, is there a way to keep the indent?

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -39,7 +39,7 @@ jobs:
         version: latest
     # Temporarily ignore the time/chrono segfault since there are no known
     # good wordarounds and they are used by dependent libraries also.
-    - run: cargo audit --ignore RUSTSEC-2020-0159 --ignore RUSTSEC-2020-0071
+    - run: cargo audit --ignore RUSTSEC-2020-0159 --ignore RUSTSEC-2020-0071 --ignore RUSTSEC-2023-0052
```
Comment (rogercloud) [inline .github/workflows/ci.yml:42]: Have we checked the previous two? Are they fixed.

File: src/async_fuse/memfs/cache/mod.rs (hunk 5 above)
Comment (rogercloud) [reply src/async_fuse/memfs/cache/mod.rs:382]: Why are they still here and unresolved?

File: .github/workflows/ci.yml (hunk 7 above)
Comment (xiaguan) [reply .github/workflows/ci.yml]: #403 , RUSTSEC-2020-0071 can be resolved by upgrading the versions of rust-s3 and some other libraries.

Comment (xiaguan) [reply .github/workflows/ci.yml]: RUSTSEC-2020-0159 is outdated.

**Review by rogercloud — APPROVED** [review]

**Review by GTwhy — APPROVED** [review]

**Pull request merged**